lib/thredded/collection_to_strings_with_cache_renderer.rb

```diff
@@ -95,10 +95,11 @@ def render_partials(view_context, collection:, render_threads:, **opts)
     # @param view_context
     # @return [Array<String>]
     def render_partials_serial(view_context, collection, opts)
-      partial_renderer = ActionView::PartialRenderer.new(@lookup_context)
-      collection.map { |object| render_partial(partial_renderer, view_context, opts.merge(object: object)) }
+      collection.map do |object|
+        partial_renderer = ActionView::PartialRenderer.new(@lookup_context, locals: { post: object })
```

Comment:
This is not always `post`, we need a cleaner fix for this

Reply:
Yes. Agreed. It seems to work in my manual testing, but I'm not sure if we can infer that based on the `object.class` or whether more work needs to be done. I didn't try with leaving it `opts.merge(object: object)`.

Reply:
@rickychilcott I think you can either do

```
partial_renderer = ActionView::PartialRenderer.new(@lookup_context, object: object) 
```

or as you say do below
```
partial_renderer.render(opts[:partial], view_context, opts.merge(object: object).body
```

Reply:
But if I'm following correctly the API of `PartialRenderer` has broken backward compatibility in 6.1:

it requires a second parameter, but in 6.0- requires only one parameter because the initialize calls `super` and AbstractRenderer.initialize requires only 1 param.

* [https://github.com/rails/rails/blob/v6.0.0/actionview/lib/action_view/renderer/partial_renderer.rb#L290] 
* [https://github.com/rails/rails/blob/v6.0.0/actionview/lib/action_view/renderer/abstract_renderer.rb#L23]

If so this then requires some kind of switch based on rails version

Reply:
```ruby
collection.map do |object|
  partial_renderer = ActionView::PartialRenderer.new(@lookup_context, locals: { post: object })
  partial_renderer.render(opts[:partial], view_context, opts).body
end
```
didn't break anything, but then we're still hardcoding the `post`


However:
```ruby
collection.map do |object|
  partial_renderer = ActionView::PartialRenderer.new(@lookup_context, object: object)
  partial_renderer.render(opts[:partial], view_context, opts).body
end
```

doesn't work. Its expecting post to be a local, and it's not there. So... this doesn't appear to fix it.

Since I'm pretty fuzzy about what's going on here, and then on top of that trying you support all these versions of Rails, I'm feeling a bit out of my element. Someone who is more familiar should be able to take this all the way.